### PR TITLE
Casper proxy caller

### PR DESCRIPTION
**Pull request by zie1ony**

### CODE REVIEW

File: examples/src/docs/events.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ pub struct PartyStarted {
 #[odra::module]
 impl PartyContract {
     #[odra(init)]
-    pub fn init(&self) {
+    pub fn init(&mut self) {
```
Comment (kpob) [inline examples/src/docs/events.rs]: Why  `mut`?

File: odra-casper/types/Cargo.toml [hunk 2]
```diff
@@ -13,4 +13,4 @@ categories = ["wasm", "smart contracts"]
 casper-types = { workspace = true }
 odra-types = { version = "0.4.0", path = "../../types" }
 casper-event-standard = { workspace = true }
-num-traits = "0.2.14"
+num-traits = { version = "0.2.14", default-features = false }
```
Comment (kpob) [inline odra-casper/types/Cargo.toml:16]: Does it use `std` by default?

File: odra-casper/proxy-caller/src/lib.rs [hunk 3]
```diff
@@ -1,56 +1,90 @@
...
+    pub runtime_args: RuntimeArgs,
+    pub attached_value: Option<U512>
+}
 
-    if let Some(amount) = attached_value {
-        let cargo_purse = create_purse();
-        let main_purse = get_main_purse();
-        transfer_from_purse_to_purse(main_purse, cargo_purse, amount, None).unwrap_or_revert();
+/// Load proxy call arguments from the runtime.
+pub fn load_args() -> ProxyCall {
```
Comment (kpob) [inline odra-casper/proxy-caller/src/lib.rs]: Maybe convert to a constructor `ProxyCall::new()`

File: odra-casper/proxy-caller/src/lib.rs [hunk 4]
```diff
@@ -1,56 +1,90 @@
...
+    ProxyCall {
+        contract_package_hash,
+        entry_point_name,
+        runtime_args,
+        attached_value
+    }
+}
 
-        args.insert("cargo_purse", cargo_purse).unwrap_or_revert();
+pub fn attach_cspr(proxy_call: &mut ProxyCall) {
```
Comment (kpob) [inline odra-casper/proxy-caller/src/lib.rs]: Maybe convert to a constructor `ProxyCall::attach_cspr(&self mut)`

**Review by kpob — APPROVED** [review]

File: examples/src/docs/events.rs (hunk 1 above)
Comment (zie1ony) [reply examples/src/docs/events.rs]: fixed

File: odra-casper/types/Cargo.toml (hunk 2 above)
Comment (zie1ony) [reply odra-casper/types/Cargo.toml:16]: yes

File: odra-casper/proxy-caller/src/lib.rs (hunk 3 above)
Comment (zie1ony) [reply odra-casper/proxy-caller/src/lib.rs]: I made it into constructor.

File: odra-casper/proxy-caller/src/lib.rs (hunk 4 above)
Comment (zie1ony) [reply odra-casper/proxy-caller/src/lib.rs]: I merged it with constructor.

**Pull request merged**